src/sentry/workflow_engine/processors/action.py

```diff
@@ -127,19 +124,26 @@ def filter_recently_fired_workflow_actions(
     Returns actions associated with the provided DataConditionsGroups, excluding those that have been recently fired. Also updates associated WorkflowActionGroupStatus objects.
     """
 
-    data_condition_group_actions = DataConditionGroupAction.objects.filter(
-        condition_group__in=filtered_action_groups
-    ).values_list("action_id", "condition_group__workflowdataconditiongroup__workflow_id")
+    data_condition_group_actions = (
+        DataConditionGroupAction.objects.filter(condition_group__in=filtered_action_groups)
+        .select_related("condition_group__workflowdataconditiongroup__workflow")
+        .values_list(
+            "action_id",
+            "condition_group__workflowdataconditiongroup__workflow_id",
+            "condition_group__workflowdataconditiongroup__workflow__config",
+        )
+    )
 
     action_to_workflows_ids: dict[int, set[int]] = defaultdict(set)
-    workflow_ids: set[int] = set()
+    workflow_frequencies: dict[int, timedelta] = {}
 
-    for action_id, workflow_id in data_condition_group_actions:
+    for action_id, workflow_id, workflow_config in data_condition_group_actions:
```

Comment:
Is it possible to type `data_condition_group_actions`? I don't think `workflow_config` should ever be `None`?